ac-core/src/main/java/es/ucm/fdi/ac/Analysis.java

```diff
@@ -150,6 +134,34 @@ public void loadSources(SourceSet sources) throws IOException {
 		}
 	}
 
+	private void addSubmissionRecursive(FileTreeNode tn,
+			HashMap<String, Submission> unique, int i) {
+		if (tn.getFile().isDirectory()) {
+			for (FileTreeNode fn : tn.getLeafChildren()) {
+				addSubmissionRecursive(fn, unique, i);
```

Comment:
if i is incremented inside recursive calls, these increments will be invisible once the calls return. This results in duplicate ids being assigned, which is bad. Given tha we have the `unique` hashmap, we can dispose with i, and just use the hashmap's current size as unique identifiers. In any case, as it is, this code is buggy.